eZ/Publish/Core/Limitation/ObjectStateLimitationType.php

```diff
@@ -126,6 +126,11 @@ public function evaluate(APILimitationValue $value, APIUserReference $currentUse
             throw new InvalidArgumentException('$object', 'Must be of type: Content, VersionInfo or ContentInfo');
         }
 
+        // Skip evaluating for RootLocation
+        if (0 === $object->id) {
```

Comment:
Might be a tiny bit cleaner to match against `mainLocationId === 1` as it's not a id hard coded in the code base.
ref: https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Repository/Helper/DomainMapper.php#L352-L370

Reply:
@ViniTou is it possible to match again location with ID = 1?

Reply:
@andrerom 
@lserwatka 
There is no problem with evaluating this against `mainLocationId`, but this `id` is hardcoded in that same place, just three lines higher, so don't know where the gain is.

Unless I am reading something wrong.
https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Repository/Helper/DomainMapper.php#L359
vs
https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Repository/Helper/DomainMapper.php#L356

Reply:
Then it is fine to check it against $object->id === 0. I would leave it as it is.

Reply:
It's not a big difference, just that locationId 1 comes from database.

_Context: We have root node location in db, but right now we don't have the content data for it as we don't want people to be able to edit it_

Reply:
Ok, context cleared me this up, thanks :D
I am just not sure, how `locationId `comes from db, unless you are saying about checking it before extracting `ContenInfo` from provided object (`Content` or `VersionInfo`) but that seems a little bit risky for me. 

i.e.
```
       if ($object instanceof Content) {
        if (1 === $object->id) {
            return true;
            $object = $object->getVersionInfo()->getContentInfo();
        }
```

Reply:
@andrerom 
I thought about it during evening and I think i got your point finally. You were talking about case, when we move this hardcoded peace of data (of `ContentInfo`) into db. Then, the `mainLocationId` would stay the same as it is code right now as it is connected to already existing `Location`.
If this is not the case, I give up. If it is, I would gladly change it because it make a lot of sense.

Reply:
Correct 😃 👍  So it's very narrow/nitpick case, and it's more about thinking ahead just in case we one day deal with that.